### PR TITLE
cors: added invalid/valid stats to cors filter

### PR DESCRIPTION
**Pull request by dschaller**
*Description*: Added stats to the cors filter for invalid vs valid origin headers.
*Risk Level*: Medium
*Testing*: Unit tests were updated to include assertions around stat invocations.
*Docs Changes*: N/A
*Release Notes*: Included

### CODE REVIEW

**Comment by mattklein123** [comment]
@codesuki do you have time to take a first pass?

**Comment by dschaller** [comment]
Thanks for the reviews @codesuki and @mattklein123!

@mattklein123 I updated the version history and cors docs in the latest push.

**Comment by codesuki** [comment]
Totally ignored the docs 😅
Will think of it next time.

File: docs/root/intro/version_history.rst [hunk 1]
```diff
@@ -36,6 +36,7 @@ Version history
 * tracing: added support to the Zipkin tracer for the :ref:`b3 <config_http_conn_man_headers_b3>` single header format.
 * upstream: changed how load calculation for :ref:`priority levels<arch_overview_load_balancing_priority_levels>` and :ref:`panic thresholds<arch_overview_load_balancing_panic_threshold>` interact. As long as normalized total health is 100% panic thresholds are disregarded.
 * upstream: changed the default hash for :ref:`ring hash <envoy_api_msg_Cluster.RingHashLbConfig>` from std::hash to `xxHash <https://github.com/Cyan4973/xxHash>`_.
+* cors: added :ref: `invalid/valid stats <cors-statistics>` to filter.
```
Comment (mattklein123) [inline docs/root/intro/version_history.rst]: please alpha sort

File: source/extensions/filters/http/cors/cors_filter.h [hunk 2]
```diff
@@ -9,9 +11,42 @@ namespace Extensions {
 namespace HttpFilters {
 namespace Cors {
 
+/**
+ * All CORS filter stats. @see stats_macros.h
+ */
+// clang-format off
+#define ALL_CORS_STATS(COUNTER)\
```
Comment (mattklein123) [inline source/extensions/filters/http/cors/cors_filter.h]: nit: right align `\` s like other examples

File: test/extensions/filters/http/cors/cors_filter_test.cc [hunk 3]
```diff
@@ -25,10 +26,13 @@ namespace Cors {
 
 class CorsFilterTest : public testing::Test {
 public:
-  CorsFilterTest() : filter_() {
```
Comment (mattklein123) [inline test/extensions/filters/http/cors/cors_filter_test.cc]: up to you, but as an aside, you can avoid many of the changes in this file by still using initializer list init. Something like this:

```
CorsFilterTest() : config_(new CorsFilterConfig("test.", stats_), filter_(config_) 
```

And move `stats_` definition above `config_`

**Comment by mattklein123** [comment]
@dschaller sorry can you check clang-tidy, it's a new check and it's asking you to fix code you didn't write but if you don't mind that would be great. 

Also, friendly request to please not rebase once you open a PR (see contributing guide). It's much harder to review. Just merge/push once the PR is opened and we will squash at the end. Thanks!

**Pull request merged**